src/System.Diagnostics.Debug/src/System/Diagnostics/DebuggerVisualizerAttribute.cs

```diff
@@ -0,0 +1,98 @@
+    [System.Runtime.InteropServices.ComVisible(true)]
+    public sealed class DebuggerVisualizerAttribute : Attribute
+    {
+        private string _visualizerObjectSourceName;
+        private string _visualizerName;
+        private Type _target;
+
+        public DebuggerVisualizerAttribute(string visualizerTypeName)
+        {
+            _visualizerName = visualizerTypeName;
+        }
+        public DebuggerVisualizerAttribute(string visualizerTypeName, string visualizerObjectSourceTypeName)
+        {
+            _visualizerName = visualizerTypeName;
+            _visualizerObjectSourceName = visualizerObjectSourceTypeName;
+        }
+        public DebuggerVisualizerAttribute(string visualizerTypeName, Type visualizerObjectSource)
+        {
+            if (visualizerObjectSource == null)
+            {
+                throw new ArgumentNullException("visualizerObjectSource");
+            }
+            _visualizerName = visualizerTypeName;
+            _visualizerObjectSourceName = visualizerObjectSource.AssemblyQualifiedName;
+        }
+        public DebuggerVisualizerAttribute(Type visualizer)
+        {
+            if (visualizer == null)
+            {
+                throw new ArgumentNullException("visualizer");
+            }
+            _visualizerName = visualizer.AssemblyQualifiedName;
+        }
+        public DebuggerVisualizerAttribute(Type visualizer, Type visualizerObjectSource)
+        {
+            if (visualizer == null)
+            {
+                throw new ArgumentNullException("visualizer");
+            }
+            if (visualizerObjectSource == null)
+            {
+                throw new ArgumentNullException("visualizerObjectSource");
+            }
+            _visualizerName = visualizer.AssemblyQualifiedName;
+            _visualizerObjectSourceName = visualizerObjectSource.AssemblyQualifiedName;
+        }
+        public DebuggerVisualizerAttribute(Type visualizer, string visualizerObjectSourceTypeName)
+        {
+            if (visualizer == null)
+            {
+                throw new ArgumentNullException("visualizer");
+            }
+            _visualizerName = visualizer.AssemblyQualifiedName;
+            _visualizerObjectSourceName = visualizerObjectSourceTypeName;
+        }
+
+        public string VisualizerObjectSourceTypeName
+        {
+            get { return _visualizerObjectSourceName; }
+        }
+        public string VisualizerTypeName
+        {
+            get { return _visualizerName; }
+        }
+        public string Description { get; set; }
+        
+        public Type Target
+        {
+            set
+            {
+                if (value == null)
+                {
+                    throw new ArgumentNullException("value");
+                }
+
+                TargetTypeName = value.AssemblyQualifiedName;
+                _target = value;
+            }
+
+            get { return _target; }
```

Comment:
Nit: It's a little strange to have the setter come before the getter. I'd move the `get` up to the top.